server/raft.go

```diff
@@ -200,6 +201,8 @@ type raft struct {
 	hcommit   uint64 // The commit at the time that applies were paused
 	pobserver bool   // Whether we were an observer at the time that applies were paused
 
+	inflightSnapshot bool // Indicates a snapshot is inflight, i.e. it's in the apply queue. Blocks us from becoming candidate.
```

Comment:
Think this needs to be a counter, otherwise more than one snapshot on the ApplyQ would do the wrong thing.

Reply:
Done.
Also have added a test for when we get sent snapshots via `SendSnapshot`, which didn't up the counter before.